class NotExhaustive(Exception):
    pass


failed = object()
succeeded = object()
